### ISSUE TITLE
Ignore calc() expressions in number-leading-zero

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#1682]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I propose to add an extra option for `number-leading-zero` to ignore numbers with leading zeros in `calc()`.

> Which rule, if any, is this issue related to?

`number-leading-zero`

> What CSS is needed to reproduce this issue?

```css
div {
    max-width: calc(700px * 0.9);
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
    "rules": {
        "number-leading-zero": "never"
    }
}
```

> Which version of stylelint are you using?

`7.0.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

Expecting no error, because it's not stylistic thing, but math.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warning was flagged:

```shell
2:26    ⚠  Unexpected leading zero (number-leading-zero) [stylelint]
```

**Comment by davidtheclark** [comment]
Why?

**Comment by hudochenkov** [comment]
`calc` expression is a mathematical expression, not just a value or a parameter. I never seen skipped leading zeros in math. It's weird for me to see this:

```css
div {
    max-width: calc(700px * .9);
}
```

**Comment by evilebottnawi** [comment]
Pretty weird to use a different style in function and styles.

**Comment by evilebottnawi** [comment]
@hudochenkov maybe always use `"number-leading-zero": "never"`?

**Comment by evilebottnawi** [comment]
@hudochenkov how about this `rgba(32,128,202,0.32)` ?

**Comment by hudochenkov** [comment]
> maybe always use `"number-leading-zero": "always"` be good solution?

Maybe.

> how about this `rgba(32,128,202,0.32)`?

It's not a math expression, it's a list of parameters.

**Comment by evilebottnawi** [comment]
@hudochenkov I do not think it's appropriate for the core, best solution is plugin.

**Comment by ntwb** [comment]
> > how about this rgba(32,128,202,0.32)?
> It's not a math expression, it's a list of parameters.

The first three are _decimal_ values of colors from the RGB color model, the fourth is the transparency **value** between `0` and `1`. 

Finding a concise definition from [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#rgba()), [Wikipedia](https://en.wikipedia.org/wiki/RGBA_color_space) was quite elusive the best I could find was [W3S](http://www.w3schools.com/cssref/css_colors_legal.asp):

> _"An RGBA color value is specified with: rgba(red, green, blue, alpha). The alpha parameter is a number between 0.0 (fully transparent) and 1.0 (fully opaque).)"_

I'm of the opinion that the fourth alpha transparency figure **is indeed** a mathematical expression of a numerical value between `0` and `1`

**Comment by hudochenkov** [comment]
Still `calc()` the only function which accept mathematical expressions with symbols like `+ – / *`. In such expressions leading zero isn't dropped.

**Comment by davidtheclark** [comment]
@hudochenkov I think your preference here is idiosyncratic enough that it makes a better plugin than core rule. If you build a plugin that is widely adopted or we hear from other people with a similar preference, we'd reconsider integrating the option into core.

@jeddy3 : For issues where we decide it should be a plugin, do you prefer leaving them open with labels, or closing them?

**Comment by jeddy3** [comment]
> For issues where we decide it should be a plugin, do you prefer leaving them open with labels, or closing them?

Label them with just ["plugin" and close](https://github.com/stylelint/stylelint/issues?q=is%3Aissue+label%3A%22type%3A+plugin%22+is%3Aclosed). 

https://github.com/stylelint/stylelint/issues/1500 & https://github.com/stylelint/stylelint/issues/1575 are only open because they're part of the `7.0.0` [release planning](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/release-planning.md). We can close them too once the dust settles :)
